app/src/test/java/org/simple/clinic/sync/ModelSyncTest.kt

```diff
@@ -36,32 +51,39 @@ class ModelSyncTest {
     val modelSync = modelSyncProvider(syncCoordinator)
 
     modelSync.sync()
-        .test()
-        .await()
-        .assertError(RuntimeException::class.java)
+        .onErrorComplete()
+        .blockingAwait()
 
     assertThat(pullCompleted).isTrue()
   }
 
   @Test
   @Parameters(method = "sync models that both push and pull")
   fun <T : Any, P : Any> `errors during pull should not affect push`(
-      modelSyncProvider: (SyncCoordinator) -> ModelSync
+      modelSyncProvider: (SyncCoordinator) -> ModelSync,
+      supportedSyncOps: Set<SyncOp>
   ) {
-    val dataSync = mock<SyncCoordinator>()
+    if ((PUSH in supportedSyncOps).not()) {
+      return
+    }
+
+    val syncCoordinator = mock<SyncCoordinator>()
     var pushCompleted = false
 
-    whenever(dataSync.pull(any<SynceableRepository<T, P>>(), any(), any())).thenReturn(Completable.error(RuntimeException()))
-    whenever(dataSync.push(any<SynceableRepository<T, P>>(), any()))
+    whenever(syncCoordinator.pull(any<SynceableRepository<T, P>>(), any(), any())).thenReturn(Completable.error(RuntimeException()))
+    whenever(syncCoordinator.push(any<SynceableRepository<T, P>>(), any()))
         .thenReturn(Completable.complete().doOnComplete { pushCompleted = true })
 
-    val modelSync = modelSyncProvider(dataSync)
+    val modelSync = modelSyncProvider(syncCoordinator)
 
     modelSync.sync()
-        .test()
-        .await()
-        .assertError(RuntimeException::class.java)
+        .onErrorComplete()
+        .blockingAwait()
 
     assertThat(pushCompleted).isTrue()
   }
+
+  enum class SyncOp {
```

Comment:
Can we use full `SyncOperation` instead? Can this also be private?

Reply:
Renamed it to `SyncOperation`. Making it private is not possible since it is a parameter of the public test methods.